Assets/_Main/Bruno/Scripts/RangedAI.cs

```diff
@@ -0,0 +1,122 @@
+using UnityEngine;
+using System.Collections.Generic;
+using System.Collections;
+
+public class RangedAI : Entity
+{
+    [SerializeField] private List<Transform> patrolPoints;
+    private int index;
+    private float distanceBeetweenPoints;
+    private float patrolRange = 1f;
+    private float rangeToDriveAway = 8f; // this variable is used to set the max range that AI has before driving away from the player
+    private float attackRange = 8.7f;
+
+    void Start()
+    {
+        playerDebug = GameObject.FindWithTag("Player").GetComponent<DebugPlayer>();
+        rigidBody = GetComponent<Rigidbody>();
+        damage = 0f;
+        speed = 13.4f;
+        index = 0;
+    }
+
+
+    void Update()
+    {
+
+        if(!playerSeen)
+        {
+            if(!idle)
+            {
+                StartCoroutine(IdleToMove());
+                SwitchBehaviour(STATE.PATROL);
+            }
+        }
+
+        if(playerSeen)
+        {
+            SwitchBehaviour(STATE.CHASE);
+        }
+
+    }
+
+
+    void OnTriggerEnter(Collider other)
+    {
+        if(other.gameObject.CompareTag("Player"))

```

Comment:
Adesso va bene il tag come debug ma aggiungi un commento per dire che il tag in fase definitiva non verrà più utilizzato.